### PR TITLE
Fix deployment issues on iOS

### PR DESCRIPTION
**Pull request by takla21**
GitHub Issue: #

## Proposed Changes

 - [x] Bug fix
 - [ ] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build or CI related changes
 - [ ] Documentation content changes
 - [ ] Other, please describe:

## Description

In this PR, I've made sure that we can compile on iOS and it doesn't crash for iOS 16 & 15.

## Impact on version

- [ ] **Major**
  - The template structure was changed.
- [ ] **Minor**
  - New functionalities were added.
- [x] **Patch**
  - A bug in behavior was fixed.
  - Documentation was changed.

### Always applicable
No matter your changes, these checks always apply.
- [x] Your conventional commits are aligned with the **Impact on version** section.
- [ ] Updated [CHANGELOG.md](../CHANGELOG.md).
  - Use the latest Major.Minor.X header if you do a **Patch** change.
  - Create a new Major.Minor.X header if you do a **Minor** or **Major** change.
  - If you create a new header, it aligns with the **Impact on version** section and matches what is generated in the build pipeline.
- [x] Documentation files were updated according with the changes.
- [x] Images about the template are referenced from the [wiki](https://github.com/nventive/UnoApplicationTemplate/wiki/Images) and added as images in this git.
- [x] TODO comments are hints for next steps for users of the template and not planned work.

### Contextual
Based on your changes these checks may not apply.
- [ ] Automated tests for the changes have been added/updated.
- [ ] Tested on all relevant platforms

### CODE REVIEW

File: src/app/ios/Podfile.lock
Comment (Soap-141) [inline src/app/ios/Podfile.lock:1]: holy shit lol

Comment (takla21) [reply src/app/ios/Podfile.lock:1]: 🤷

File: src/app/ios/Podfile.lock [hunk 1]
```diff
@@ -1,23 +1,192 @@
...
+  package_info_plus: 58f0028419748fad15bf008b270aaa8e54380b1c
+  PromisesObjC: f5707f49cb48b9636751c5b2e7d227e43fba9f47
   shared_preferences_foundation: b4c3b4cddf1c21f02770737f147a3f5da9d39695
+  url_launcher_ios: 6116280ddcfe98ab8820085d8d76ae7449447586
 
-PODFILE CHECKSUM: abf3b855940916c3c7a39484bfae1567c819589a
+PODFILE CHECKSUM: 495df3bf13e5006823c35b0807e801ca9a64427c
 
-COCOAPODS: 1.15.2
+COCOAPODS: 1.13.0
```
Comment (Soap-141) [inline src/app/ios/Podfile.lock:192]: Just hope it doesn't break CI, we use 1.15.2

**Review by Soap-141 — APPROVED** [review]

**Review by Marc-Antoine-Soucy — APPROVED** [review]

File: src/app/ios/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -455,7 +454,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 17.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 15.0;
```
Comment (jeanplevesque) [inline src/app/ios/Runner.xcodeproj/project.pbxproj:457]: So IPHONEOS_DEPLOYMENT_TARGET is for the minimum version?

Comment (takla21) [reply src/app/ios/Runner.xcodeproj/project.pbxproj:457]: Yes it is according to this https://docs.flutter.dev/deployment/ios#review-xcode-project-settings

> iOS Deployment Target
>    The minimum iOS version that your app supports. Flutter supports iOS 12 and later. If your app or plugins include >Objective-C or Swift code that makes use of APIs newer than iOS 12, update this setting to the highest required version.

File: src/app/ios/Podfile.lock (hunk 1 above)
Comment (takla21) [reply src/app/ios/Podfile.lock:192]: The CI seems fine

**Pull request merged**